### ISSUE TITLE
Widgets removed - Android Domoticz subscription

### ISSUE THREAD

**Issue by Rouzax** [domoticz/domoticz-android#713]
I wanted to install a new phone, and I am trying to set up my Domoticz widgets again, but every time I do the Domoticz app will start crashing each time I'm trying to open it.
Looking at the GitHub page it would appear that Widgets have been removed.
[https://github.com/domoticz/domoticz-an ... 6fbc1aa277](https://github.com/domoticz/domoticz-android/commit/a9366a6f7ea2ded910c270efbbf8416fbc1aa277)

Is that correct, are widgets removed and not supported anymore?

**Comment by galadril** [comment]
Yes this is related. We need completely rebuild the widgets with the latest Android sdk. There are also issues with beacon scanning etc.

 So hoping to get a new app update approval from google soon.

**Comment by Rouzax** [comment]
> Yes this is related. We need completely rebuild the widgets with the latest Android sdk. There are also issues with beacon scanning etc.
> 
> So hoping to get a new app update approval from google soon.

So it should only be temporarily?

_Just tested with the old Domoticz Premium app which I also still have access to and there it still works._

**Comment by Rouzax** [comment]
@galadril, is this removal temporary and what timeframe are you looking to reimplement widgets again?
The widgets and geofence are the primary reason I pay for a subscription.

**Comment by Rouzax** [comment]
@galadril, can you please let me know how long you think widgets will be removed for?
Main reason for my Premium subscription is widgets an geofence

**Comment by galadril** [comment]
No ETA, its a lot of work

**Comment by galadril** [comment]
@Rouzax I'm finally getting somewhere with the widgets 

![image](https://github.com/domoticz/domoticz-android/assets/14561640/a51097bf-54fa-4b1e-a230-5af88f84a434)

![image](https://github.com/domoticz/domoticz-android/assets/14561640/a876c116-5aba-4357-b086-4b1ecdd16e6b)

I think i even found the bug which caused sometimes the widgets to loose their state when updating the app.
Also the storage of the data per widget is completely rewritten from shareprefs to sqlite db

https://github.com/domoticz/domoticz-android/pull/715

**Comment by Rouzax** [comment]
Great to hear, is it already rolled into the app?

**Comment by l1gi** [comment]
I am able to see the widgets and able to pick one device, but then I will get an error that

`Widgets are not allowed on this device, please, check Domoticz app settings.`

If I touch the button 'AKTUALIZACE' there I am redirected to Domoticz settings, but there is NO widgets options as far as I can see.

What am I looking for? Any advice, please?

![Screenshot_20240725_085157_Domoticz](https://github.com/user-attachments/assets/4130b713-f71c-4c8f-bb41-3bf08d400280)
